Lib/test/test_external_inspection.py

```diff
@@ -458,24 +514,32 @@ async def main():
                 self.assertIn(('Task-1', []), entries)
                 self.assertIn(('server task', [[['main'], 'Task-1', []]]), entries)
                 self.assertIn(('echo client spam', [[['main'], 'Task-1', []]]), entries)
+
+                expected_stack = [[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]]
+                tasks_with_stack = [task for task in entries if task[1] == expected_stack]
+                self.assertGreaterEqual(len(tasks_with_stack), 1000)
+
                 # the final task will have some random number, but it should for
-                # sure be one of the echo client spam horde
-                self.assertEqual([[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]], entries[-1][1])
+                # sure be one of the echo client spam horde (In windows this is not true
+                # for some reason)
+                if sys.platform != "win32":
```

Comment:
@ambv I don't have any idea why this is not true on windows, but apart from some mysterious final task with no stack the thing is the same as linux and macos

Reply:
This is what entries has:

```
...
 ('Task-2000',
  [[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]]),
 ('Task-2002',
  [[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]]),
 ('Task-2004',
  [[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]]),
 ('Task-2006',
  [[['echo_client_spam'], 'echo client spam', [[['main'], 'Task-1', []]]]]),
 ('Task-2009', [])]
```

Reply:
Perhaps the different default asyncio loop on Windows is having an impact? You might get consistent results if you force it back to the selector loop instead of the proactor loop (I think those are the right terms... I'm a bit out of date on asyncio, tbqh)